src/main/scala/org/broadinstitute/dsde/firecloud/service/MethodsService.scala

```diff
@@ -1,21 +1,86 @@
+import org.broadinstitute.dsde.firecloud.model.{HttpResponseWithErrorReport, ErrorReport}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.{FireCloudConfig}
+import org.slf4j.{ LoggerFactory}
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
 
 class MethodsServiceActor extends Actor with MethodsService {
   def actorRefFactory = context
   def receive = runRoute(routes)
 }
 
-trait MethodsService extends HttpService with FireCloudDirectives {
+
+trait MethodsService extends HttpService with PerRequestCreator with FireCloudDirectives {
 
   lazy val log = LoggerFactory.getLogger(getClass)
 
-  val routes: Route =
-    passthroughAllPaths("methods", FireCloudConfig.Agora.methodsBaseUrl) ~
-    passthroughAllPaths("configurations", FireCloudConfig.Agora.configurationsBaseUrl)
+  // Agora permissions that require special handling
+
+  val methodsAndConfigsACLOverrideRoute: Route =
+    path( "configurations|methods".r / Segment / Segment / IntNumber / "permissions") { ( agora_ent, namespace, name, snapshotId) =>
+      val url = getUrlFromBasePath(agora_ent, namespace, name, snapshotId)
+      get { requestContext =>
+        // pass to AgoraPermissionHandler
+        perRequest(requestContext,
+          Props(new AgoraPermissionActor(requestContext)),
+          AgoraPermissionHandler.Get(url))
+      } ~
+        post {
+          // take the body of the HTTP POST and construct a FireCloudPermission from it
+          entity(as[List[FireCloudPermission]]) {
+            fireCloudPermissions => requestContext =>
+              //scan user and roles of each input to make sure they're valid
+              //this validityFlag represents a list of booleans where each item
+              //is true if an individual input is valid and okay to send to Agora
+              val validityFlags=fireCloudPermissions.map(x => x.verifyValidUserAndRole)
+              //see http://stackoverflow.com/questions/22518773/applying-logical-and-to-list-of-boolean-values
+              val andAllByUsersAndRoles=validityFlags.foldLeft(true)(_ && _)
+              if(!(andAllByUsersAndRoles))
+                {
+                requestContext.complete(HttpResponseWithErrorReport(BadRequest,
+                  "Invalid user or access setting detected! Each permission posted"+
+                    " must have a non-empty user and a permission of READER, OWNER, or NO ACCESS!"))
+                }
+
+              // translate the FireCloudPermissions into  AgoraPermissions
+              val agoraPermissions = fireCloudPermissions.map(x => x.toAgoraPermission)
+              // pass to AgoraPermissionHandler
+              perRequest(
+                requestContext,
+                Props(new AgoraPermissionActor(requestContext)),
+                AgoraPermissionHandler.Post(url, agoraPermissions))
+          }
```

Comment:
So ... if you find yourself in scala-land creating lots of vals and using if conditions, those are classic signs of java-thinking. This whole method could be simplified to something like this:
```
fireCloudPermissions match {
	case x if x.map(_.verifyValidUserAndRole).forall _ =>
	  perRequest(
		requestContext,
		Props(new AgoraPermissionActor(requestContext)),
		AgoraPermissionHandler.Post(url, fireCloudPermissions.map(_.toAgoraPermission)))
	case _ =>
	  RequestCompleteWithErrorReport(BadRequest,
		"Invalid user or access setting detected! Each permission posted"+
		  " must have a non-empty user and a permission of READER, OWNER, or NO ACCESS!")
}
```
I'm not a fan of the `verifyValidUserAndRole` and `toAgoraPermission` methods ... let me think on that one for a bit.